DS13/code/modules/borg/items+structures.dm

```diff
@@ -0,0 +1,557 @@
+	name = "Subspace vector translocator"
+	desc = "A device which opens a transwarp conduit to rapidly translate a unimatrix across subspace through a vector. Click it with an open hand to set a target, then walk/run into it to engage translocation (run right into the middle of it)."
+	icon = 'DS13/icons/borg/large_machines.dmi'
+	icon_state = "teleporter-off"
+	var/turf/open/target
+	resistance_flags = INDESTRUCTIBLE | LAVA_PROOF | FIRE_PROOF | UNACIDABLE | ACID_PROOF | FREEZE_PROOF
+	var/teleport_sound = 'DS13/sound/effects/borg/borg_transporter.ogg'
+	anchored = TRUE
+	can_be_unanchored = FALSE
+	density = TRUE
+	pixel_x = -32
+
+/obj/structure/borg_teleporter/examine(mob/user)
+	. = ..()
+	to_chat(user, "<I> Translocation matrix assigned: Unimatrix 325, grid 006. [get_area(target)]</I>")
+
+/obj/structure/borg_teleporter/Bumped(atom/movable/AM)
+	if(!target)
+		return ..()
+	if(isliving(AM))
+		new /obj/effect/temp_visual/dir_setting/ninja/cloak(get_turf(AM), AM.dir)
+		do_sparks(1, 8, src)
+		visible_message("[AM] enters a transwarp conduit!")
+		AM.forceMove(target)
+		playsound(AM.loc, teleport_sound, 100,1)
+		new /obj/effect/temp_visual/dir_setting/ninja/cloak(get_turf(AM), AM.dir)
+
+/obj/structure/borg_teleporter/attack_hand(mob/user)
+	. = ..()
+	icon_state = "teleporter-off"
+	target = null
+	var/A
+	A = input(user, "Set Translocation Target", "Translocation shift matrix", A) as null|anything in GLOB.teleportlocs
+	var/area/thearea = GLOB.teleportlocs[A]
+	var/list/L = list()
+	for(var/turf/T in get_area_turfs(thearea.type))
+		if(!is_blocked_turf(T))
+			L += T
+	target = pick(L)
+	if(!target)
+		to_chat(user, "Unable to set translocation target, could not locate a suitable subspace exit point.")
+		return
+	to_chat(user, "Coordinates locked: [target.x],[target.y]. Unimatrix 325, grid 006")
+	icon_state = "teleporter-on"
+
+//Borg tool
+
+/obj/item/borg_tool
+	name = "Primary prosthetic interaction node"
+	desc = "This massive prosthetic houses a vicious claw and a full replicator suite for the rapid conversion of an area into God knows what. Click it while holding it to change its mode."
+	icon = 'DS13/icons/weapons/melee.dmi'
+	item_state = "borgtool"
+	icon_state = "borgtool"
+	lefthand_file = 'DS13/icons/mob/inhands/weapons/guns_lefthand.dmi'
+	righthand_file = 'DS13/icons/mob/inhands/weapons/guns_righthand.dmi'
+	resistance_flags = UNACIDABLE
+	var/mode = "assimilate"
+	var/convert_time = 30 //3 seconds
+	item_flags = NODROP
+	force = 18 //hella strong
+	var/removing_airlock = FALSE
+	var/dismantling_machine = 0
+	var/blacklistedmachines = list(/obj/machinery/computer/communications, /obj/machinery/computer/card)
+	var/saved_time
+	var/cooldown = 10
+	var/resource_amount = 10 //Starts with a bit so you can build a structure from the get-go
+	var/resource_cost = 10
+	var/busy = FALSE //Are you performing an action already? This avoids infinite turf conversion
+
+/obj/item/borg_tool/attack_self(mob/user)
```

Comment:
We will need to work out a better method some people have had issues with it. (We should have combat be automatic on non borg mobs).

Reply:
Perhaps, but it's really not that hard to grasp.